src/notices/cc-warning-payment.js

```diff
@@ -0,0 +1,127 @@
+import { ERROR_TYPES } from '@clevercloud/client/esm/utils/payment.js';
+import { css, html, LitElement } from 'lit-element';
+import { i18n } from '../lib/i18n.js';
+import { defaultThemeStyles } from '../styles/default-theme.js';
+import { linkStyles } from '../templates/cc-link.js';
+
+const warningSvg = new URL('../assets/warning.svg', import.meta.url).href;
+
+/**
+ * @typedef {import('./types.js').PaymentMethodError} PaymentMethodError
+ * @typedef {import('./types.js').PaymentWarningModeType} PaymentWarningModeType
+ */
+
+/**
+ * A component to display a warning block with details about payment methods errors.
+ *
+ * @cssdisplay grid
+ */
+export class CcWarningPayment extends LitElement {
+
+  static get properties () {
+    return {
+      errors: { type: Array },
+      mode: { type: String },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {Array<PaymentMethodError>} */
+    this.errors = [{ type: ERROR_TYPES.NO_PAYMENT_METHOD }];
```

Comment:
* We do use: `type[]` instead of `Array<type>` in our other components we do need to change it otherwise type definitions plugin doesn't recognize it. 
* Type lacks a description
* You need to export your interface in your type file

Reply:
Good catch.